### PR TITLE
Friends fragment

### PR DESCRIPTION
**Pull request by gracewainaina**

### CODE REVIEW

**Comment by gracewainaina** [comment]
Here is a demo for the new people fragment which displays user names, profile photo and music bios of users https://github.com/gracewainaina/Musicca/blob/friends-fragment/Screenshots/people_fragment_demo.mp4

File: Musicca/app/src/main/java/com/example/musicca/adapters/PeopleAdapter.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+public class PeopleAdapter extends RecyclerView.Adapter<PeopleAdapter.ViewHolder> {
+
+    private Context context;
+    private List<ParseUser> users;
+
+    public PeopleAdapter(Context context, List<ParseUser> users) {
+        this.context = context;
+        this.users = users;
+        Log.i("PEOPLE ADAPTER", "ADAPTER created ");
+        Log.i("PEOPLE ADAPTER", "ADAPTER users: " + this.users.size());
```
Comment (xuyueee) [inline Musicca/app/src/main/java/com/example/musicca/adapters/PeopleAdapter.java]: Remove logs please.

File: Musicca/app/src/main/java/com/example/musicca/fragments/PeopleFragment.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            public void onRefresh() {
+                // Remember to CLEAR OUT old items before appending in the new ones
+                peopleAdapter.clear();
+                queryUsers(0);
+                peopleAdapter.notifyDataSetChanged();
+            }
+        });
+    }
+
+    protected void queryUsers(int page) {
```
Comment (xuyueee) [inline Musicca/app/src/main/java/com/example/musicca/fragments/PeopleFragment.java]: What is `page` supposed to do? It's not used. 
Also, on this fragment, do you show all the users in the database?

File: Musicca/app/src/main/java/com/example/musicca/adapters/PeopleAdapter.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+            tvUsername = itemView.findViewById(R.id.tvUsername);
+            ivProfileImage = itemView.findViewById(R.id.ivProfileImage);
+            tvMusicBio = itemView.findViewById(R.id.tvMusicBio);
+        }
+
+        public void bind(ParseUser parseUser) {
+            tvUsername.setText(parseUser.getUsername());
+            tvMusicBio.setText(parseUser.getString("musicbio"));
+            ParseFile profileImage = parseUser.getParseFile("profileimage");
```
Comment (xuyueee) [inline Musicca/app/src/main/java/com/example/musicca/adapters/PeopleAdapter.java]: No string literals please.

**Review by xuyueee — COMMENTED** [review]
The demo looks good! I like that you can see people's music bios.

File: Musicca/app/src/main/java/com/example/musicca/fragments/PeopleFragment.java (hunk 2 above)
Comment (gracewainaina) [reply Musicca/app/src/main/java/com/example/musicca/fragments/PeopleFragment.java]: Yes, I show all the users in the database in this fragment. 
I also got rid of the int page, I realized I was no longer using it

File: Musicca/app/src/main/java/com/example/musicca/fragments/PeopleFragment.java [hunk 4]
```diff
@@ -70,20 +70,24 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+            public void done(List<ParseUser> parseUsers, ParseException e) {
                 if (e == null) {
                     // The query was successful.
-                    allUsers.addAll(objects);
-
+                    // removing current user from the list
+                    for (int i = 0; i < parseUsers.size(); i++) {
+                        if (parseUsers.get(i).getUsername() != ParseUser.getCurrentUser().getUsername()){
+                            allUsers.add(parseUsers.get(i));
+                        }
```
Comment (gracewainaina) [inline Musicca/app/src/main/java/com/example/musicca/fragments/PeopleFragment.java:89]: I also ensure that current user is not added to the list of users being bound by the adapter to the fragment's recycler view

**Pull request merged**